packages/node_modules/@webex/webex-core/test/unit/spec/services/interceptors/server-error.js

```diff
@@ -0,0 +1,161 @@
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import sinon from 'sinon';
+import {
+  ServerErrorInterceptor,
+  WebexHttpError
+} from '@webex/webex-core';
+
+const {assert} = chai;
+
+chai.use(chaiAsPromised);
+sinon.assert.expose(chai.assert, {prefix: ''});
+
+describe('webex-core', () => {
+  describe('ServerErrorInterceptor', () => {
+    let interceptor;
+
+    before(() => {
+      interceptor = new ServerErrorInterceptor();
+    });
+
+    describe('#onResponseError()', () => {
+      let options;
+      let reason;
+
+      beforeEach(() => {
+        options = {};
+      });
+
+      describe('when reason is a webex server error and the uri exist', () => {
+        beforeEach(() => {
+          options.uri = 'http://not-a-url.com/';
+          reason = new WebexHttpError.InternalServerError({
+            message: 'test message',
+            statusCode: 500,
+            options: {
+              url: 'http://not-a-url.com/',
+              headers: {
+                trackingId: 'tid'
+              }
+            }
+          });
+        });
+
+        describe('when the web-ha feature is enabled', () => {
+          let get;
+          let markUrlFailedAndGetNew;
+          let submitClientMetrics;
+
+          beforeEach('mock device and metrics', () => {
+            interceptor.webex = {
+              internal: {
+                device: {
+                  features: {
+                    developer: {
+                      get: sinon.stub()
+                    }
+                  },
+                  markUrlFailedAndGetNew: sinon.stub()
+                },
+                metrics: {
+                  submitClientMetrics: sinon.spy()
+                }
+              }
+            };
+
+            get = interceptor.webex.internal.device.features.developer.get;
+            markUrlFailedAndGetNew = interceptor
+              .webex
+              .internal
+              .device
+              .markUrlFailedAndGetNew;
+            submitClientMetrics = interceptor
+              .webex
+              .internal
+              .metrics
+              .submitClientMetrics;
+
+            get.returns({value: true});
+            markUrlFailedAndGetNew.resolves();
+          });
+
+          it('should get the feature \'web-high-availability\'', () =>
+            interceptor.onResponseError(options, reason)
+              .then(() => assert.isTrue(false, 'returned a resolved promise'))
```

Comment:
I'm assuming this line is here to indicate the test failure. Instead of putting in tests assertions we don't plan on using, could we use `done` instead inside the `catch`? That way if the test actually fails, it will just timeout and report that error. Something like:

```js
it('should get the feature \'web-high-availability\'', (done) => {
  interceptor.onResponseError(options, reason)
    .catch(() => {
      assert.calledWith(get, 'web-high-availability');
      done();
  });
});
```

Reply:
Implemented.